### ISSUE TITLE
What is the GPU type you trained with?

### ISSUE THREAD

**Issue by peirongxu** [tijiang13/InstantAvatar#22]
Different types of GPU have different training speeds. Since fast training is the biggest advantage of InstantAvatar, GPU type should be clarified. But I cannot find this info in the repo or your paper.
Actually, we trained with a NVIDIA 3090 and it costed 30 mins at least, far slower than 60 secs. Could you tell me your GPU type? Or there is any other reason for expectation mismatch?

**Comment by tijiang13** [comment]
Hi Peirong,

The test was done on a single 3090, which is already mentioned in the paper. We show the the 1 minute training setting in the "bash/run-demo.sh", where we train the model for 30 epochs (it should be around 40-50 iters per second ~ 2s per epoch).

Best,
Tianjian

**Comment by peirongxu** [comment]
> Hi Peirong,
> 
> The test was done on a single 3090, which is already mentioned in the paper. We show the the 1 minute training setting in the "bash/run-demo.sh", where we train the model for 30 epochs (it should be around 40-50 iters per second ~ 2s per epoch).
> 
> Best, Tianjian

Sorry, I ignored the setting in the paper. But I got the max speed of 4.31 it/s, not 40-50 it/s. An epoch costs average 25s.
It's very strange. What is your GPU usage when training a model?

**Comment by tijiang13** [comment]
Hi Peirong,

I checked the bash script once again and I guess that's where the 30mins come [bash/run-neuman-demo](https://github.com/tijiang13/InstantAvatar/blob/master/bash/run-neuman-demo.sh)?

You should be able to achieve similar result if you fit the model for 100 epochs. I somehow end up uploading the debug script...

Best,
Tianjian

**Comment by tijiang13** [comment]
I just run the test quickly on my machine (there is some debugging code but it should only make the training slower), and that's what I got.

<img width="1467" alt="image" src="https://github.com/tijiang13/InstantAvatar/assets/15152165/b8d47f12-a7f5-4a9c-a31b-b5db00751b29">

Best,
Tianjian

**Comment by tijiang13** [comment]
And nvidia-smi shows following:

<img width="566" alt="image" src="https://github.com/tijiang13/InstantAvatar/assets/15152165/8b7cc0ad-b317-4ffc-8a51-a9aa097d71fb">

Best, Tianjian

**Comment by peirongxu** [comment]
> > > Hi Peirong,
> > > The test was done on a single 3090, which is already mentioned in the paper. We show the the 1 minute training setting in the "bash/run-demo.sh", where we train the model for 30 epochs (it should be around 40-50 iters per second ~ 2s per epoch).
> > > Best, Tianjian
> > 
> > 
> > Sorry, I ignored the setting in the paper. But I got the max speed of 4.31 it/s, not 40-50 it/s. An epoch costs average 25s. It's very strange. What is your GPU usage when training a model?
> 
> That's weird... Are you running it on some server?
> 
> Best,Tianjian

Yes, I am running the code on an IceLake 8358 with Nvidia 3090. But the server is only running a process at the same time, without other users or processes. Is it possible that low cpu frequency makes training slow?

**Comment by tijiang13** [comment]
> > > > Hi Peirong,
> > > > The test was done on a single 3090, which is already mentioned in the paper. We show the the 1 minute training setting in the "bash/run-demo.sh", where we train the model for 30 epochs (it should be around 40-50 iters per second ~ 2s per epoch).
> > > > Best, Tianjian
> > > 
> > > 
> > > Sorry, I ignored the setting in the paper. But I got the max speed of 4.31 it/s, not 40-50 it/s. An epoch costs average 25s. It's very strange. What is your GPU usage when training a model?
> > 
> > 
> > That's weird... Are you running it on some server?
> > Best,Tianjian
> 
> Yes, I am running the code on an IceLake 8358 with Nvidia 3090. But the server is only running a process at the same time, without other users or processes. Is it possible that low cpu frequency makes training slow?

It could be CPU load and slow disk I/O. Currently, the dataloader reads images and masks from disk during each iteration, which can be slow when using distributed storage or slower HDDs. 

You can try increasing the number of threads for the dataloader, it should be able to mitigate IO latency with more threads doing the prefetch.

Best,
Tianjian

**Comment by peirongxu** [comment]
OK, I found it because of server CPU which led to long time cost during an iteration. And it made GPU usage very low(~20%). So that I got slow training speed.
Thanks for your enthusiastic help! You can close the issue^_^
